dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/AppStartTimeProvider.kt

```diff
@@ -0,0 +1,15 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.internal
+
+internal interface AppStartTimeProvider {
+    /**
+     * Provide the application start time in nanoseconds from device boot, or our best guess if
```

Comment:
> from device boot
From application boot, no?

Also, can we elaborate on the "our best guess" with the actual implementation? 
i.e. from RUM class load in Android < N

Reply:
I can yes. I left it vague so that if it changes we wouldn't have to worry about updating the documentation or have it be wrong.

Reply:
Also, the time is actually nanoseconds since device boot (it's using `[System.nanoTime](https://developer.android.com/reference/java/lang/System#nanoTime())` or the `elapsedRealtime` functions, which are both timed from JVM or device boot time.

Reply:
Ok, now I understand what you meant by "from device boot".
Initially I interpreted it as "we are returning the time difference between the app started and the time you booted your device"